src/ORM/FieldType/DBBoolean.php

```diff
@@ -71,16 +73,16 @@ public function scaffoldSearchField($title = null)
             0 => _t(__CLASS__ . '.NOANSWER', 'No')
         ];
 
-        return (new DropdownField($this->name, $title, $source))
+        return DropdownField::create($this->name, $title, $source)
             ->setEmptyString($anyText);
     }
 
-    public function nullValue()
+    public function nullValue(): ?int
     {
         return 0;
     }
 
-    public function prepValueForDB($value)
+    public function prepValueForDB(mixed $value): array|int|null
```

Comment:
This method must _always_ allow returning an array so that subclasses can do funky custom stuff. See `MockDynamicAssignmentDBField` in tests for an example of this.